### PR TITLE
fix: Respect globalDependencies when determining changed packages

### PR DESCRIPTION
**Pull request by NicholasLYang**
### Description

We didn't respect the `globalDependencies` field in `turbo.json` when determining changed packages. This meant that you could have a global dependency changed, but turbo wouldn't run.

### Testing Instructions

Closes TURBO-2486

### CODE REVIEW

**Review by chris-olszewski — CHANGES_REQUESTED** [review]
Needs tests

File: turborepo-tests/integration/tests/filter-run.t [hunk 1]
```diff
@@ -26,4 +26,18 @@ Setup
    Tasks:    0 successful, 0 total
   Cached:    0 cached, 0 total
     Time:\s*[\.0-9]+m?s  (re)
-  
\ No newline at end of file
+  
+
```
Comment (chris-olszewski) [inline turborepo-tests/integration/tests/filter-run.t:30]: This works as an integration test, but we have pretty good fixtures in the `filter.rs` unit tests and feels like we could get away with just those.

**Review by chris-olszewski — APPROVED** [review]
I would prefer unit tests, but if this passes all tests, then I'm fine with it as is.

File: turborepo-tests/integration/tests/filter-run.t (hunk 1 above)
Comment (NicholasLYang) [reply turborepo-tests/integration/tests/filter-run.t:30]: From a quick scan, I'm not sure we have unit tests that specifically handle git ranges where you can specify the packages changed. TestChangeDetector works by storing the changed packages for a git ref range, not the changed files.

**Pull request merged**